source/language/classical.rst

```diff
@@ -427,6 +427,278 @@ Terminating the program early
+      brace-enclosed scope.  Declarations within a switch statement but
+      outside of a ``case`` or ``default`` statement are ill-formed.
+
+	  A ``case`` or ``default`` statement may create brace-enclosed scope,
+	  if such statements use the optional left and right brace literals.
+
+	  ``case`` statements may create scope by using both left and
+      right braces, or shall not use braces at all, in which case no
+      scope is created.
+
+	  ``default`` statements may create scope by using both left and
+      right braces, or shall not use braces at all, in which case no
+      scope is created.
+
+	  A ``break`` statement shall be followed by a semicolon literal (;).
+      ``break`` statements not followed by a semicolon literal shall raise
+      an error diagnostic.
+
+	  A ``break`` statement followed by a semicolon terminates the ``case``
+      or ``default`` statement it belongs to.
+
+	  Declarations at brace-enclosed ``case`` or ``default`` scope shall
+      be in scope only within the scope of their brace enclosing ``case``
+      or ``default`` statements, regardless of whether the enclosing
+      statement contains a terminating ``break`` statement or not.
+
+	  Declarations of types that automatically acquire global scope in
+      OpenQASM3 - such as gates, qubits and defcals - are not allowed
+      at ``case`` or ``default`` ``switch`` statement scope. Use of such
+	  declarations is ill-formed and requires a compiler diagnostic.
+
+	  Duplicate values for ``case`` statements controlling labels are not
+      allowed. The compiler shall issue an error diagnostic in such cases.
+
+	  A ``case`` or ``default`` statement ending with a ``break`` statement
+      followed by a semicolon terminates the execution of the ``switch``
+      statement. After executing all the statements of the ``case`` or
+      ``default`` statement, control flow is then transferred to the first
+      statement following the closing right brace of the enclosing
+      ``switch`` statement.
+
+	  A ``case`` statement that does not contain a terminating ``break``
+      statement followed by a semicolon transfers control to the
+      statement immediately following it.
+
+	  A ``default`` statement that does not contain a terminating ``break``
+      statement followed by a semicolon transfers control outside the
+      enclosing ``switch`` statement, to the first statement immediately
+      following.
+
+      This is sometimes referred to as "fall-through".
+
+      The compiler shall issue a warning diagnostic for all fall-through
+      ``case`` or ``default`` ``switch`` statements.
+
+	  A ``switch`` statement shall contain at least one ``case`` statement.
+      A ``switch`` statement with no ``case`` statements shall raise an
+      error diagnostic.
+
+	  A ``switch`` statement is not required to contain a ``default`` statement.
+      In this case, the compiler shall issue a warning diagnostic.
+
+  Examples:
+
+.. code-block::
```

Comment:
In front of each example can we please add a short sentence describing what it is demonstrating?

Reply:
Yes it looks like my merge got all messed-up so I will have to re-format and re-submit this anyway.